presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java

```diff
@@ -92,6 +92,8 @@ public ThriftHiveRecordCursorProvider(HdfsEnvironment hdfsEnvironment, ThriftFie
 
         return Optional.of(new ThriftHiveRecordCursor<>(
                 genericRecordReader(recordReader),
+                path,
+                start,
```

Comment:
Oh I see. I didn't expand, thought the exception was in the constructor. I see it's in advanceNextPosition.